actors/builtin/multisig/multisig_actor.go

```diff
@@ -60,114 +62,188 @@ func (a *MultiSigActor) State(rt Runtime) (vmr.ActorStateHandle, MultiSigActorSt
 	return h, state
 }
 
-func (a *MultiSigActor) Propose(rt vmr.Runtime, txn MultiSigTransaction) TxnID {
+type ConstructorParams struct {
+	AuthorizedParties  autil.ActorIDSetHAMT
+	NumApprovalsThreshold int64
+}
+
+func (a *MultiSigActor) Constructor(rt vmr.Runtime, params *ConstructorParams) {
+
+	rt.ValidateImmediateCallerIs(builtin.InitActorAddr)
+	h := rt.AcquireState()
+
+	st := MultiSigActorState{
+		AuthorizedParties:     params.AuthorizedParties,
+		NumApprovalsThreshold: params.NumApprovalsThreshold,
+		PendingTxns:           MultiSigTransactionHAMT_Empty(),
+		PendingApprovals:      MultiSigApprovalSetHAMT_Empty(),
+	}
+
+	UpdateRelease_MultiSig(rt, h, st)
+}
+
+type ProposeParams struct {
+	To addr.Address
+	Value big.Int
+	Method abi.MethodNum
+	Params abi.MethodParams
+}
+
+func (a *MultiSigActor) Propose(rt vmr.Runtime, params ProposeParams) TxnID {
 	vmr.RT_ValidateImmediateCallerIsSignable(rt)
 	callerAddr := rt.ImmediateCaller()
 	a._rtValidateAuthorizedPartyOrAbort(rt, callerAddr)
 
 	h, st := a.State(rt)
 	txnID := st.NextTxnID
 	st.NextTxnID += 1
+
+	txn := MultiSigTransaction{
+		Proposer:   callerAddr,
+		Expiration: 0, // TODO lotus does not specify this.
+		To:         params.To,
+		Method:     params.Method,
+		Params:     params.Params,
+		Value:      0, // TODO lotus does not specify this.
+	}
```

Comment:
https://github.com/filecoin-project/lotus/blob/https://github.com/filecoin-project/lotus/commit/8418464d91d8eb7e5a2765668073579291912dbb/chain/actors/actor_multisig.go#L196

Reply:
See https://github.com/filecoin-project/specs-actors/issues/5

I suggest including Expiration in the propose params for now so it's internally consistent.

Value should be set to the value or rt.ValueReceived() and *not* be an explicit parameter, sounds like a bug in Lotus (cc @whyrusleeping)